### PR TITLE
Add user deleted event (Lombiq Technologies: OCORE-35)

### LINKED ISSUE #6719: Add user deleted event
In `OrchardCore.Users.Handlers.IUserEventHandler` there is a `CreatedAsync` but no `DeletedAsync`.

It should be added with an empty interface default implementation to preserve compatibility. It should be invoked from `OrchardCore.Users.Services.UserStore` like its counterpart for created event.

### PR DESCRIPTION
**Pull request by DAud-IcI**
Fixes #6719

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=6720) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=6720) before we can accept your contribution.<br/><br/>:x: DAud-IcI [sign now](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=6720)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OrchardCMS/OrchardCore?pullRequest=6720) it.</sub>

**Comment by agriffard** [comment]
I changed the title and description for you.
When you do a PR, write `Fixes #IssueNumber` to close the isssue when the PR is merged.

**Comment by agriffard** [comment]
You might also want to add a Workflows/Drivers/UserDeletedEventDisplay, Views/Items/UserDeletedEvent.Fields.Design.cshtml, UserDeletedEvent.Fields.Edit.cshtml and UserDeletedEvent.Fields.Thumbnail.cshtml this in Workflows/startup.cs:

`            services.AddActivity<UserCreatedEvent, UserCreatedEventDisplay>();`
for the visual Workflow event.

Look at what has been done for the UserCreatedEvent.

**Comment by DAud-IcI** [comment]
Thanks, I've added them.

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Design.cshtml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@model OrchardCore.Users.Workflows.ViewModels.UserEventViewModel<OrchardCore.Users.Workflows.Activities.UserDeletedEvent>
+
+<header>
+    <h4><i class="fa fa-user"></i>@Model.Activity.GetTitleOrDefault(() => T["User Deleted"])</h4>
```
Comment (agriffard) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Design.cshtml:4]: `fa-user-minus` could be appropriate.
https://fontawesome.com/icons/user-minus?style=solid

Comment (DAud-IcI) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Design.cshtml]: Very nice, changed.

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Thumbnail.cshtml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<h4 class="card-title"><i class="fa fa-user"></i>@T["User Deleted"]</h4>
```
Comment (agriffard) [inline src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Thumbnail.cshtml:1]: Please use same icon everywhere.

Comment (DAud-IcI) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Thumbnail.cshtml:1]: Ok.

**Comment by DAud-IcI** [comment]
Is there anything else needed for this to be merged?

**Comment by agriffard** [comment]
Did you actually try it in a Workflow?

**Comment by DAud-IcI** [comment]
Yes, it worked all right. I created just a simple one that logs a string when a user is deleted:
image

File: src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Design.cshtml (hunk 1 above)
Comment (DAud-IcI) [reply src/OrchardCore.Modules/OrchardCore.Users/Views/Items/UserDeletedEvent.Fields.Design.cshtml]: It just occurred to me that I should do the same with fa-user-plus for UserCreatedEvent. Pushed.

**Pull request merged**